### PR TITLE
Add SaveAllCommand

### PR DESCRIPTION
**Pull request by diluculo**
Please see #215 and #237

### CODE REVIEW

File: src/Gemini/Modules/Shell/Commands/SaveAllFilesCommandHandler.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        {
+            var tasks = new List<Task<Tuple<IPersistedDocument, bool>>>();
+
+            foreach (var document in _shell.Documents)
+            {
+                var persistedDocument = document as IPersistedDocument;
+                if (persistedDocument == null) continue;
+
+                // skip if the file is new
+                if (!persistedDocument.IsNew)
```
Comment (tgjones) [inline src/Gemini/Modules/Shell/Commands/SaveAllFilesCommandHandler.cs]: Is this what Visual Studio does?

**Comment by mwpowellhtx** [comment]
For my edification, is "what" (this) what Visual Studio does?

**Comment by diluculo** [comment]
@tgjones Actually the SaveAll command in VS saves all edited (IsDirty) documents and it shows "Save Prompt" for new documents. However, I think "skip new document" is enough, if users know it.

**Comment by mwpowellhtx** [comment]
Yes, I believe that's correct, including project changes.

**Comment by tgjones** [comment]
LGTM - @diluculo is this ready to merge?

**Comment by diluculo** [comment]
Please merge this, which was rebased onto the latest master.

**Comment by tgjones** [comment]
Thanks!

**Pull request merged**